meteor/server/cronjobs.ts

```diff
@@ -90,20 +91,38 @@ async function restartCasparCG(systemSettings: ICoreSystemSettings | undefined,
 	let shouldRetryAttempt = false
 	const ps: Array<Promise<any>> = []
 
-	const casparcgAndParentDevices = (await PeripheralDevices.findFetchAsync(
-		{
-			type: PeripheralDeviceType.PLAYOUT,
-			subType: { $in: [PERIPHERAL_SUBTYPE_PROCESS, TSR.DeviceType.CASPARCG] },
-		},
-		{
-			projection: {
-				_id: 1,
-				subType: 1,
-				parentDeviceId: 1,
-				lastSeen: 1,
+	const [casparcgAndParentDevices, activePlaylists] = await Promise.all([
+		PeripheralDevices.findFetchAsync(
+			{
+				type: PeripheralDeviceType.PLAYOUT,
+				subType: { $in: [PERIPHERAL_SUBTYPE_PROCESS, TSR.DeviceType.CASPARCG] },
 			},
-		}
-	)) as Array<Pick<PeripheralDevice, '_id' | 'subType' | 'parentDeviceId' | 'lastSeen'>>
+			{
+				projection: {
+					_id: 1,
+					subType: 1,
+					parentDeviceId: 1,
+					lastSeen: 1,
+					studioAndConfigId: 1,
+				},
+			}
+		) as Promise<
+			Array<Pick<PeripheralDevice, '_id' | 'subType' | 'parentDeviceId' | 'lastSeen' | 'studioAndConfigId'>>
+		>,
+		RundownPlaylists.findFetchAsync(
+			{
+				activationId: {
+					$exists: true,
+				},
+			},
```

Comment:
_This is perhaps nit-picking, as most/all Sofie installations we know of are single-studio at the moment, but since the data model allows for multi-studio:_

This query will fail to match any [indexes we've set up in MongoDB](https://github.com/nrkno/sofie-core/blob/6e04a7f53cbe088c3db88bdc394675c27b456706/meteor/server/collections/rundown.ts#L144). Could we instead first query for the PeripheralDevices, and then use their studioId to filter in the query?
```
{
  studioId: {$in: relevantStudioIds },
  activationId: { $exists: true }
}
```

Reply:
Yeah, I guess I wanted to be fancy, but I'm good with doing this in series.